experiments/experiment.py

```diff
@@ -157,6 +157,8 @@ def format_data(self, state, full_state=True):
                 data[key] = float(data[value])
             elif isinstance(value, dict):
                 data[key] = self.format_data(value)
+            elif isinstance(value, (list, set)):
+                data[key] = list(map(lambda x: float(x), value))
```

Comment:
and here, we shouldn't be auto-casting to `float`, let's use a recursive call to `format_data`:
```suggestion
                data[key] = list(map(lambda x: self.format_data(x), value))
```